src/main/groovy/com/deploygate/gradle/plugins/utils/AndroidPlatformUtils.groovy

```diff
@@ -0,0 +1,106 @@
+package com.deploygate.gradle.plugins.utils
+
+import org.gradle.api.Project
+
+class AndroidPlatformUtils {
+    private static AGPVersion AGP_VERSION
+
+    static String getAapt2Location(Project project) {
+        return project.deploygate.aapt2Path ?: System.getenv('DEPLOYGATE_APPT2_PATH') ?: "${getSdkHome(project)}/build-tools/${getBuildToolsVersion(project)}/aapt2"
+    }
+
+    static AGPVersion getAGPVersion() {
+        if (!AGP_VERSION) {
+            AGP_VERSION = AndroidPlatformUtils.AGPVersion.parse(getVersionString())
+        }
+
+        return AGP_VERSION
+    }
+
+    static boolean isAppBundleSupported() {
+        return getAGPVersion().major >= 3 && getAGPVersion().minor >= 2
+    }
+
+    private static String getVersionString() {
+        try {
+            return Class.forName("com.android.builder.model.Version").getField("ANDROID_GRADLE_PLUGIN_VERSION").get(null)
+        } catch (Throwable ignored) {
+        }
+
+        // before 3.1
+        try {
+            return Class.forName("com.android.builder.Version").getField("ANDROID_GRADLE_PLUGIN_VERSION").get(null)
+        } catch (Throwable ignored) {
+        }
+
+        return Integer.MAX_VALUE + ".0.0"
+    }
+
+    private static String getBuildToolsVersion(Project project) {
+        try {
+            def buildToolRevision = Class.forName('com.android.repository.Revision').getMethod('safeParseRevision', String.class).invoke(null, project.android.buildToolsVersion)
+
+            def klass = Class.forName('com.android.builder.core.AndroidBuilder')
+
+            if (buildToolRevision < klass.getField('MIN_BUILD_TOOLS_REV').get(null)) {
+                buildToolRevision = klass.getField('DEFAULT_BUILD_TOOLS_REVISION').get(null)
+            }
+
+            return buildToolRevision
+        } catch (Throwable ignored) {
+            // print error
+            return null
+        }
+    }
+
+    private static String getSdkHome(Project project) {
```

Comment:
You can retrieve it from `AppExtension#sdkDirectory` (= `project.android.sdkDirectory`)

https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.AppExtension.html#com.android.build.gradle.AppExtension:sdkDirectory

Reply:
I didn't know that and that was what I really wanted! NOTED.